### PR TITLE
feat: initiate holdings response call

### PR DESCRIPTION
**Pull request by rhnvrm**

### CODE REVIEW

File: portfolio.go [hunk 1]
```diff
@@ -122,3 +122,41 @@ func (c *Client) ConvertPosition(positionParams ConvertPositionParams) (bool, er
 
 	return b, err
 }
+
+// HoldingsAuthParams represents the input params for initiating holdings authorization
+type HoldingsAuthParams struct {
+	ISIN     string
+	Quantity float64
```
Comment (vividvilla) [inline portfolio.go:129]: There are more params like 

`type`, `transfer_type`, `counter_demat`, `exec_date` so this struct had to include those and have another struct to hold `isin` and `quantity` fields which can be repeated.

Something like below 

```
type HoldingsAuthIns struct {
	ISIN     string
	Quantity float64
}

type HoldingsAuthParams struct {
   Ins []HoldingsAuthIns
   ....
   ...
}
```

**Review by vividvilla — CHANGES_REQUESTED** [review]
Also I feel we can return the redirect URL keeping it as a constant.

**Comment by rhnvrm** [comment]
> Also I feel we can return the redirect URL keeping it as a constant.

Resolved in https://github.com/zerodha/gokiteconnect/pull/45/commits/083896a943dd327bc33e304858e8450ff2d5718c

File: portfolio.go (hunk 1 above)
Comment (rhnvrm) [reply portfolio.go:130]: Added these fields

File: portfolio.go [hunk 2]
```diff
@@ -122,3 +122,71 @@ func (c *Client) ConvertPosition(positionParams ConvertPositionParams) (bool, er
...
+// use within the holdings auth initialization.
+type HoldingsAuthInstruments struct {
+	ISIN     string
+	Quantity float64
+}
+
+// HoldingAuthParams represents the inputs for initiating holdings authorization.
+type HoldingAuthParams struct {
+	Type         string
+	TransferType string
```
Comment (vividvilla) [inline portfolio.go:136]: We can add constants for `TransferType` 

```go
	TransferTypePreTrade  = "pre"
	TransferTypePostTrade = "post"
	TransferTypeOffMarket = "off"
	TransferTypeGift      = "gift"
```

Comment (vividvilla) [reply portfolio.go:136]: and for `Type` as well

```
	TypeMF     = "mf"
	TypeEquity = "equity"
```

**Review by vividvilla — CHANGES_REQUESTED** [review]

File: portfolio.go (hunk 2 above)
Comment (vividvilla) [reply portfolio.go:136]: Prefix the consts accordingly.

Comment (rhnvrm) [reply portfolio.go:146]: Added in 51022e4

**Pull request merged**